js/lex.go

```diff
@@ -66,21 +90,34 @@ func (tt TokenType) String() string {
 
 // Lexer is the state for the lexer.
 type Lexer struct {
-	r *buffer.Lexer
-
-	regexpState   bool
-	templateState bool
+	r     *buffer.Lexer
+	stack []ParsingContext
```

Comment:
I think `context` would be fine, as indexed as `context[i]` it would read as `get context at index i`. Reading `stack[i]` leaves one wondering what kind of stack it is. I would also call it `Context` instead of `ParsingContext` because it is pretty obvious that it is for parsing purposes as that is what the package is called.